tests/test_butler.py

```diff
@@ -723,12 +750,13 @@ def testCompositePutGetConcrete(self) -> None:
         self.assertIn("423", str(uri), f"Checking visit is in URI {uri}")
 
         # Predicted dataset
-        dataId = {"instrument": "DummyCamComp", "visit": 424}
-        uri, components = butler.getURIs(datasets[0].datasetType, dataId=dataId, predict=True)
-        self.assertFalse(components)
-        self.assertIsInstance(uri, ResourcePath)
-        self.assertIn("424", str(uri), f"Checking visit is in URI {uri}")
-        self.assertEqual(uri.fragment, "predicted", f"Checking for fragment in {uri}")
+        if self.predictionSupported:
```

Comment:
Is this a temporary restriction on remote butler or a permanent one?

Reply:
It's not currently implemented.  I don't really understand the use cases for that feature so I'm not sure whether it makes sense to implement in the future or not.

Reply:
Yes, it's a tricky one. People wanted to know what the path would be if they were to store the dataset. In client/server this becomes a bit odder since you have to decide what it comes back as since returning a signed URL to something that doesn't exist seems odd. 

We have discussed in the past whether getURI should have the option of returning the server-side URI (s3://bucket/path_in_store/file.fits) rather than the client-side signed URL -- presumably this is assumed to be a bad idea because the bucket name shouldn't leak to an untrusted user? Maybe for predicted URIs we could return something like `butler://datastore_name/path_in_store/file.fits#predicted` so that people could see what the final filename template would resolve to. I'm not sure how many users care about this at this point.

Reply:
Maybe this could become a `DirectButler`-only interface?  It certainly won't be the last one; we might even want to go ahead and move the entire write interface from `Butler` to `DirectButler` at some point.